### PR TITLE
Implements the TRIM(X) function

### PR DESCRIPTION
**Pull request by juandent**
Adds the TRIM(X) function to the functions available in sqlpp11.

Roland, please check :-)

Juan Dent

### CODE REVIEW

File: include/sqlpp11/aggregate_functions/trim.h [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+  struct trim_t : public expression_operators<trim_t<Flag, Expr>, text>,
+                   public alias_operators<trim_t<Flag, Expr>>
+  {
+    using _traits = make_traits<text, tag::is_expression, tag::is_selectable>;
+
+    using _nodes = detail::type_vector<Expr, aggregate_function>;
+    using _can_be_null = std::true_type;
+    using _is_aggregate_expression = std::true_type;
+
+    //static_assert(is_noop<Flag>::value or std::is_same<distinct_t, Flag>::value,
```
Comment (rbock) [inline include/sqlpp11/aggregate_functions/trim.h]: This comment seems to be a left-over

File: include/sqlpp11/aggregate_functions/trim.h [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    static Context& _(const T& t, Context& context)
+    {
+      context << "TRIM(";
+      serialize(t._expr, context);
+      context << ")";
+      return context;
+    }
+
+	// check for presence of flag
+#if 0
```
Comment (rbock) [inline include/sqlpp11/aggregate_functions/trim.h]: I would omit this and the other `if 0` block

**Comment by rbock** [comment]
Hi Juan,

Thanks for the pull request!

As you see there seems to be a  conflict in the parameter list. Not sure what happened there, yet.

I suggest, that you do not mix the two things (trim and parameter list). It would be simpler to handle those two things in separate pull requests.

Regarding the trim function: I would omit the two parameter version completely for the time being.

File: include/sqlpp11/aggregate_functions/trim.h (hunk 2 above)
Comment (juandent) [reply include/sqlpp11/aggregate_functions/trim.h]: ok yep.

File: include/sqlpp11/aggregate_functions/trim.h (hunk 1 above)
Comment (juandent) [reply include/sqlpp11/aggregate_functions/trim.h]: Yes, I wasn't sure if we needed any of that... in a different way perhaps (I am aware distinct has nothing to do with TRIM)... but yes, we should delete that!

**Comment by juandent** [comment]
I agree - forget the parameter_list, it was something I had done and it stayed uncommitted. Not sure how to remove it from the pull request?

**Comment by rbock** [comment]
You can do a `git reset --soft` to where you started with your commits, drop the parameter-list change, and create a new pull request.

Again, github has cool documentation and a bit of googling will help you find the best way to do it (I am rarely creating pull requests and therefore not the best direct source of knowledge for this).

File: include/sqlpp11/aggregate_functions/trim.h [hunk 3]
```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2013-2015, Roland Bock
```
Comment (rbock) [inline include/sqlpp11/aggregate_functions/trim.h]: You might want to add yourself here

File: include/sqlpp11/aggregate_functions/trim.h [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef SQLPP11_AGGREGATE_FUNCTIONS_TRIM_H
```
Comment (rbock) [inline include/sqlpp11/aggregate_functions/trim.h]: `SQLPP11_TRIM_H` would be fine here.

File: include/sqlpp11/aggregate_functions/trim.h [hunk 5]
```diff
@@ -0,0 +1,110 @@
+/*
```
Comment (rbock) [inline include/sqlpp11/aggregate_functions/trim.h]: This file should be located in `sqlpp11`, not in `sqlpp11/aggregate_functions` (I just pointed you there for examples, sorry for the confusion).

File: include/sqlpp11/aggregate_functions/trim.h [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef SQLPP11_AGGREGATE_FUNCTIONS_TRIM_H
+#define SQLPP11_AGGREGATE_FUNCTIONS_TRIM_H
+
+#include <sqlpp11/char_sequence.h>
+#include <sqlpp11/select_flags.h>
```
Comment (rbock) [inline include/sqlpp11/aggregate_functions/trim.h]: You probably do not need to include `select_flags` here.

File: include/sqlpp11/aggregate_functions/trim.h [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+#include <sqlpp11/data_types/integral/data_type.h>
```
Comment (rbock) [inline include/sqlpp11/aggregate_functions/trim.h]: I think you should include `data_types/text/data_type.h`.

File: include/sqlpp11/aggregate_functions/trim.h [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+      return context;
+    }
+
+  };
+
+  template <typename T>
+  auto trim(T t) -> trim_t<noop, wrap_operand_t<T>>
+  {
+    static_assert(not contains_aggregate_function_t<wrap_operand_t<T>>::value,
+                  "trim() cannot be used on an aggregate function");
```
Comment (rbock) [inline include/sqlpp11/aggregate_functions/trim.h]: Is it really bad if the argument of trim is an aggregate? Looks like a copy-paste left-over.

**Comment by rbock** [comment]
Closing this as outdated due to #205

**Pull request closed without merging**